sys/action.js

```diff
@@ -137,7 +137,14 @@ function buildQueryResponse(res) {
     } else if (!res.body || res.body.error) {
         throw apiError((res.body || {}).error);
     } else if (!res.body.query || (!res.body.query.pages && !res.body.query.userinfo)) {
-        throw apiError({ info: 'Missing query pages from the PHP action API response.' });
+        throw new HTTPError({
+            status: 404,
+            body: {
+                type: 'not_found#page_revisions',
```

Comment:
Hm.. Actually for normal `query` requests the `pages` property is always there, but the missing articles are marked with `missing` property in the results. However for search generator the pages is not in the response, which indicates a not found. So, I think it's semantically OK to throw a 404 here, just change the wording a bit